TestGitHubProj/Viacheslav/Circle/CircleViewController.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  CircleViewController.swift
+//  TestGitHubProj
+//
+//  Created by Viacheslav Markov on 11.10.2020.
+//
+
+import UIKit
+
+class CircleViewController: UIViewController {
+    
+    @IBOutlet weak var greenView: UIView!
+    @IBOutlet weak var redView: UIView!
+    @IBOutlet weak var blueView: UIView!
+    
+    @IBOutlet weak var topFrameView: UIView!
+    @IBOutlet weak var rightFrameView: UIView!
+    @IBOutlet weak var leadingFrameView: UIView!
+    @IBOutlet weak var leftFrameView: UIView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        cornerRadius(to: greenView)
+        cornerRadius(to: redView)
+        cornerRadius(to: blueView)
+        
+        let panGreen = UIPanGestureRecognizer(target: self, action: #selector(dragTheCicrle))
+        panGreen.name = "panGreen"
+        greenView.addGestureRecognizer(panGreen)
+        
+        let panRed = UIPanGestureRecognizer(target: self, action: #selector(dragTheCicrle))
+        panRed.name = "panRed"
+        redView.addGestureRecognizer(panRed)
+        
+        let panBlue = UIPanGestureRecognizer(target: self, action: #selector(dragTheCicrle))
+        panBlue.name = "panBlue"
+        blueView.addGestureRecognizer(panBlue)
+        
+    }
+    
+    func cornerRadius(to circleView: UIView) {
+        circleView.layer.cornerRadius = circleView.layer.frame.height / 2
+    }
+    
+    @objc func dragTheCicrle(recognaizer: UIPanGestureRecognizer) {
+        
+        var someView: UIView!
```

Comment:
Please avoid `!` in your code. This might be an issue in future. I suggested to remove `!` and after the switch statement (line 60) add this code:
`guard var someView = someView else { return }`